frontend/src/components/utils/MobileHeaderNavDropdown.tsx

```diff
@@ -160,8 +157,6 @@ const MobileNavDropdown = () => {
   useEffect(() => {
     if (isMenuShown) {
       document.addEventListener("click", closeMenu);
-    } else {
-      document.removeEventListener("click", closeMenu);
```

Comment:
Why do you believe this should be removed?

Reply:
Listener is removed on effect destructor, so every time the effect is called (as well as on unmount) - previous listener is cleaned up.